### PR TITLE
interfaces/apparmor: update apparmor template of s-u-n for changes in Go

### PR DESCRIPTION
**Pull request by bboozzoo**
Rebuilding snap-update-ns with Go 1.22.0 triggers a new denial with the process trying to read /proc/self/maps.

The denial shows up as:

```
audit: type=1400 audit(1707404561.548:1563): apparmor="DENIED"
       operation="open" class="file"
       profile="snap-update-ns.test-snapd-sh-core22"
       name="/proc/2993630/maps" pid=2993630 comm="5"
       requested_mask="r" denied_mask="r" fsuid=0 ouid=0
```

Inspecting with strace, the attempt to open /proc/self/maps happens just a handful of syscalls after execing into snap-update-ns.

Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

File: interfaces/apparmor/template.go [hunk 1]
```diff
@@ -975,6 +975,9 @@ profile snap-update-ns.###SNAP_INSTANCE_NAME### (attach_disconnected) {
   # Allow reading the command line (snap-update-ns uses it in pre-Go bootstrap code).
   @{PROC}/@{pid}/cmdline r,
 
+  # Allow reading of own maps (Go runtime)
+  @{PROC}/@{pid}/maps r,
```
Comment (bboozzoo) [inline interfaces/apparmor/template.go:979]: I'm actually surprised that we don't have `owner` here and next to the other `@{PROC}/@{pid}/` entries like we have in the template.

Comment (alexmurray) [reply interfaces/apparmor/template.go:979]: I think it makes more sense to scope this to `owner` as suggested (and we could do the same for `cmdline` above as well) to make it more explicit that this is about granting permission for snap-update-ns to read its own cmdline/maps etc.

**Review by alexmurray — CHANGES_REQUESTED** [review]
LGTM but I would prefer this to be scope to `owner` as suggested already.

File: interfaces/apparmor/template.go (hunk 1 above)
Comment (bboozzoo) [reply interfaces/apparmor/template.go]: I've pushed an update and also added `owner` to fds and cgroup (that one even had a comment about reading own cgroup).

**Review by Meulengracht — APPROVED** [review]
Thank you for this! +1

**Review by alexmurray — APPROVED** [review]
LGTM!

**Pull request merged**